### DISCUSSION TITLE
v6: Unable to get searchParams

### DISCUSSION THREAD

**Discussion by kael** [remix-run/react-router#8457] [Q&A] [answered]
With v6.0.2, I'm trying to get URL searchParams but the router rewrites the URL without searchParams.

When the user opens `/search?name=foo` she gets client-side redirected to  `/search` and `useLocation().search;` returns an empty string.

I've tested with this [example borrowed from v5](https://v5.reactrouter.com/web/example/query-parameters), and whereas it works in the codesandebox with v5 (writing the URL in the address bar allows to get searchParams with `useLocation`), it doesn't work with v6.

Am I missing something ?

Thanks.

**Comment by bar-boko** [comment, marked as answer]
The question that you need to ask yourself is what is the `<Routes>` tree that you've created and If you're using navigate during the rendering process until the app reaches the page `/search`

**Reply by kael** [reply]
It does actually work with v6.

I recreated a router and there's no redirection from `/search?name=foo` to `/search`anymore.

But it's still the case with the original code I was working on, perhaps some specific interaction between paths - but I haven't found the cause, I'd need to investigate deeper, though.

Thanks.

**Reply by mitirius** [reply]
I am having the same issue, and no I haven't used navigate in the rendering path so far, but using Outlets and nested routes.

**Reply by kael** [reply]
@mitirius Are you using nested routes and/or routes with unusual path like `<Route path="@:userId"/>` ? I had to remove nested routes and reorganize them to make URL with searchParams work.

**Reply by mitirius** [reply]
Yes, I do, just one level though. Does that effect this?

This is my Router code :
```
<Routes>
      <Route path="/" element={<AppContainer />}>
        <Route path=":readerId" element={<ReaderSelector />} />
        <Route
          path=":readerId/:startTimeParam/:endTimeParam"
          element={<ResultsPage />}
        />
        <Route path="*" element={<NoMatch />} />
      </Route>
    </Routes>
```

**Reply by kael** [reply]
@mitirius  Would you have an example of an URL with the search params called against the router ?
